Automattic/jetpack
# Social: Use global state for connections modal

Right now having the connections modal being fired from more than one place is difficult because of how the modal opening logic depends upon the local state as well as the keyring result.

## Proposed changes:

* Move the connections modal state from local to global via redux store
* Update the consumer components to use the global modal and the corresponding actions/selectors
* Restore the "Connect accounts" CTA which was removed in #37697

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






These steps should be followed for Social admin, Jetpack settings > Sharing and the editor page

* Go to connections management UI
* Confirm that everything works as before like connecting/reconnecting an account
* Confirm that the confirmation screen works as before
* Goto Social admin page
* Remove all the connections
* Confirm that you now see the "Connect accounts" CTA att he top beside "Write a post" button
* Click on it and connect an account
* Confirm that it works as expected
* Confirm that nothing is broken for simple site post editor
